core/src/main/resources/hudson/PluginManager/advanced.jelly

```diff
@@ -78,9 +78,13 @@ THE SOFTWARE.
                 <h2 class="jenkins-section__title">${%Update Site}</h2>
                 <f:form method="post" action="siteConfigure" name="siteConfigure">
                     <f:entry title="${%URL}">
-                        <f:textbox name="site"
-                                   value="${app.updateCenter.getSite(app.updateCenter.ID_DEFAULT).url}"/>
+                      <a id="reset-to-default" href="#">${%Reset to default}</a>
+                      <f:textbox name="site"
+                                 id="update-site-url"
+                                 value="${app.updateCenter.getSite(app.updateCenter.ID_DEFAULT).url}"
+                                 checkUrl="'checkUpdateSiteUrl?value='+this.value"/>
```

Comment:
this would introduce new CSP errors from my understanding.

I think you need to do something like:

```suggestion
                                 checkUrl="'checkUpdateSiteUrl" checkDependsOn=""/>
```

@daniel-beck can you check this please?

Reply:
Good point, thanks for spotting it 👍  From my understanding this should work. Manual test should confirm.

See also https://github.com/jenkins-infra/jenkins.io/pull/5301/files#diff-4405e13cac1124b62201d74e661384a7eeccc56aa8515e5678bd97503631a5a9R70-R76

Reply:
Thanks for your suggestion!
Notice that Deploy Plugin send the request in the same way. It may be addressed by opening another issue.

Reply:
>  Notice that Deploy Plugin send the request in the same way. It may be addressed by opening another issue.

Right, this is a fairly recent project without published docs, JEP, or other guidelines :) Once we know how plugin maintainers should implement such things and have it documented, we can make a case for plugins to be improved.